diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/NewlinesRule.kt

```diff
@@ -294,25 +313,46 @@ class NewlinesRule(private val configRules: List<RulesConfig>) : Rule("newlines"
     private fun ASTNode.isDotFromPackageOrImport() = elementType == DOT &&
             parent({ it.elementType == IMPORT_DIRECTIVE || it.elementType == PACKAGE_DIRECTIVE }, true) != null
 
-    /**
-     *  taking all expressions inside complex expression until we reach lambda arguments
-     */
     private fun ASTNode.isCallsChain() = getParentExpressions()
             .lastOrNull()
             ?.run {
                 mutableListOf<ASTNode>().also {
-                    getAllLeafsWithSpecificType(DOT, it)
-                    getAllLeafsWithSpecificType(SAFE_ACCESS, it)
+                    getOrderedCallExpressions(DOT, SAFE_ACCESS, it)
                 }
             }
+            ?.filterNot {
+                // should skip node's that are val params. For example: b.some(z.a()) -> skip z.a()
+                it.treeParent.treeParent != null && it.treeParent.treeParent.elementType == VALUE_ARGUMENT
```

Comment:
I think there could be other examples when dots are used differently (like type parameters with some qualified class names). I think better approach here will be to collect dots and `?.`s only from actual calls inside `getOrderedCallExpressions`: you could, for example, get `psi as KtDotQualifiedExpression` and then check is right side is qualified expression too and so on. This way you won't need to filter retrieved values afterwards.